### ISSUE TITLE
Help with audit.rules file

### ISSUE THREAD

**Issue by bkeithryder** [kemra102/puppet-auditd#2]
Hi there,

I'm trying to use your auditd module with puppet, and it's working beautifully for managing the service and the auditd.conf.  But I'm stymied trying to get any rules defined.  I even copied/pasted from your example, but my audit.rules file is coming up blank.  Am I missing something?

Here's what I've got in my manifest.  

```
class profile::hardening::audit {

# Use puppet forge module kemra102/auditd raher than rolling our own

  class { '::auditd':
    manage_service      => true,
    space_left_action   => 'email',
    num_logs            => '5',
    max_log_file        => '100',
    max_log_file_action => 'rotate',
  }

  auditd::rule { 'delete other rules':
   content => '-D',
   order   => '00',
  }
}
```

After a puppet run, my audit.rules file looks like this:

```
# This file is managed by Puppet. DO NOT EDIT.
```

I've wiggled options a number of different ways, but all my auditd::rule definitions are just being ignored.  Any thoughts?

Thanks much,
Keith

**Comment by bkeithryder** [comment]
Sorry for the odd formatting.  Looks like my comments are being interpreted as wiki headers.

**Comment by kemra102** [comment]
Hi Keith,

Thanks for reporting the issue, I'm looking into it now. As it may help can you tell me the following:

* OS & Version Used
* Puppet Version

In order to avoid the formatting issues in future you can [fenced code blocks](https://help.github.com/articles/github-flavored-markdown/#fenced-code-blocks).

Thanks
Danny

**Comment by bkeithryder** [comment]
Thanks, Danny.

I'm running a Vagrant CentOS VM on my Mac, and eventually targeting RHEL 6.6.  Haven't tried it yet on the RHEL box.  Here are the particulars on my VM:
```
[root@xmaster manifests]# uname -a
Linux xmaster.vagrant.vm 2.6.32-504.8.1.el6.x86_64 #1 SMP Wed Jan 28 21:11:36 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

[root@xmaster manifests]# cat /etc/redhat-release
CentOS release 6.6 (Final)

[root@xmaster manifests]# puppet --version
3.7.4 (Puppet Enterprise 3.7.2)
```

**Comment by swizzley** [comment]
@bkeithryder You need to pass $order as an integer, you're passing it as a string:

 auditd::rule { 'delete other rules':
    content => '-D',
    order   => 00,
  }

**Comment by kemra102** [comment]
@swizzley In my testing this makes no difference - it still fails to populate the file.

@bkeithryder I've fixed some general issues with the **init.pp** and that seems to have fixed it for your example but the example in the README throws some horrible dependency errors. Can you confirm if the latest Git master branch at least works for you?

**Comment by bkeithryder** [comment]
Thanks to both.  I agree that it makes no difference whether $order is quoted or not -- I tested both ways before posting the original question.

@kemra102 -- I'll pull the latest branch and test in the morning when I get to work.  Thanks much!

**Comment by swizzley** [comment]
@bkeithryder are you using the latest stdlib 4.6?

**Comment by swizzley** [comment]
@kemra102 also you have a typo in init
    validate_abolute_path($krb5_key_file)

**Comment by kemra102** [comment]
Thanks @swizzley fixed in latest commit.

**Comment by bkeithryder** [comment]
@kemra102  Looks like that did that trick!  Many thanks for jumping on it so quickly.

And yes, I am using the latest stdlib.

**Comment by kemra102** [comment]
@bkeithryder Thanks for confirming, I'll put that out as a bugfix and update here when I have done so. The CentOS 7 issues seems to be related to a bug in concat so unrelated to anything here.

**Comment by kemra102** [comment]
Fix published as v1.0.1 to Puppet Forge.
